File: fix_commas.js

const fs = require('fs');
const path = 'lib/data/sample_data.dart';

try {
  let content = fs.readFileSync(path, 'utf8');
  
  // Fix: ''' followed by newline and items:
  // We use a regex that looks for ''' followed by whitespace and then items:
  // and we insert a comma.
  // Regex explanation:
  // (''')       - Group 1: The closing triple quotes
  // (\s+)       - Group 2: Whitespace (newlines/spaces)
  // (items:)    - Group 3: The items key
  // Replacement: $1,$2$3
  
  // Note: We need to be careful not to double commas if they exist.
  // But strictly speaking, the user missed them.
  // A safer regex: '''(?!\s*,)\s+(items:|version:)
  
  // Pattern 1: theory ending, then items
  content = content.replace(/'''(\s+items:)/g, "''',$1");
  
  // Pattern 2: theory ending, then version
  content = content.replace(/'''(\s+version:)/g, "''',$1");
  
  fs.writeFileSync(path, content, 'utf8');
  console.log('Successfully fixed commas in sample_data.dart');
  
} catch (err) {
  console.error('Error fixing file:', err);
  process.exit(1);
}

File: fix_commas.py

import os

file_path = r'c:\Users\migue\Desktop\project2\lib\data\sample_data.dart'

with open(file_path, 'r', encoding='utf-8') as f:
    content = f.read()

# Fix missing comma before items
content = content.replace("'''\n    items:", "''',\n    items:")

# Fix missing comma before version
content = content.replace("'''\n    version:", "''',\n    version:")

# Also handle cases where there might be extra newlines or different indentation
# But based on the read output, it seems consistent. 
# Let's also check for version before theory (Level 7) which I did manually.
# Level 7: version: 7, theory: '''...''' items:
# So for Level 7, we need comma after ''' before items. Covered by first replace.
# Levels 8-15: theory: '''...''' version: 7, items:
# So we need comma after ''' before version. Covered by second replace.

with open(file_path, 'w', encoding='utf-8') as f:
    f.write(content)

print("Fixed commas in sample_data.dart")
